tests/integration/transformers/test_id.py

```diff
@@ -589,7 +589,7 @@ def assert_proportions(self, out, samples):
         assert np.isclose(one_count, two_count * 2, atol=samples * 0.2)
         assert np.isclose(one_count, three_count * 3, atol=samples * 0.2)
         assert len(out) == samples
-        assert more_count == 0
+        assert more_count <= 1
```

Comment:
why did this change?

Reply:
It's an edge case if the following happens:

 * The `regex_format` can't generate enough regexes
 * During the first reverse_transform call, not all copies of an element are generated, ie these remaining samples get stored in the `self._remaining_samples` attribute
 * Reverse gets called a second time with the same or more num_samples than the first call

In that case the second call will end up generating the same value as what had been stored in `self._remaining_samples`, so it may end up repeated more than expected.

Reply:
What is done in the cases where cardinality_rule is not scale?

Reply:
* `cardinality_rule = None`: simply loop through the generator, there are no fallback samples. `out2` would be `[a,b,a,b,a]` in this case.
* `cardinality_rule = 'unique'`: if there are not enough values remaining in the generator, reset the generator. If even that's not enough, create fallback samples based on the samples generated so far.
* `cardinality_rule = 'match'`: the first time reverse transform is called, generate all the regex values necessary to match the data cardinality seen during fit, and store it in the `_unique_regex_values` attribute. Any extra/future samples will simply be copies of the values stored in this attribute, repeated until we have enough samples.